test/functional/diff/screen_spec.lua

```diff
@@ -0,0 +1,515 @@
+local lfs = require('lfs')
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+
+local eq = helpers.eq
+local feed = helpers.feed
+local eval = helpers.eval
+local clear = helpers.clear
+local funcs = helpers.funcs
+local nvim_prog = helpers.nvim_prog
+local write_file = helpers.write_file
+
+local function nvim_prog_abs()
+  -- system(['build/bin/nvim']) does not work for whatever reason. It needs to
+  -- either be executable searched in $PATH or something starting with / or ./.
+  if nvim_prog:match('[/\\]') then
+    return funcs.fnamemodify(nvim_prog, ':p')
+  else
+    return nvim_prog
+  end
+end
+
+describe('Diff mode screen', function()
+  describe('generic', function()
+    local fname = 'Xtest-functional-diff-screen-1'
+    local fname_2 = fname .. '.2'
+
+    local withScreen = function(w, h, action)
+      screen = Screen.new(w, h)
```

Comment:
Unless there is a good need for screens of different sizes (there probably isn't), use the standard pattern i e of `test/functional/ui/searchhl_spec.lua`.
```
  local screen

  before_each(function()
    clear()
    screen = Screen.new(40, 16)
    screen:attach()
    screen:set_default_attr_ids({
      [1] = {foreground = Screen.colors.DarkBlue, background = Screen.colors.WebGray},
      -- et cetera
    })
  end)

```
Detach can be skipped, it is not needed. Also allocating all attrs in a `set_default_attr_ids` also makes the test easier to read: one can compare two `expects()` in the same file with a quick glance.

Reply:
That's handy, thanks!